packages/cozy-harvest-lib/src/components/KonnectorConfiguration/ConfigurationTab/BIContractActivationWindow.spec.jsx

```diff
@@ -16,55 +16,99 @@ jest.mock('cozy-ui/transpiled/react/Popup', () => {
     return null
   })
 })
+jest.mock('../../InAppBrowser', () => {
+  return jest.fn().mockImplementation(({ onClose }) => {
+    setTimeout(onClose, 1)
+    return null
+  })
+})
+
+jest.mock('cozy-device-helper')
 import { findKonnectorPolicy } from '../../../konnector-policies'
 import Popup from 'cozy-ui/transpiled/react/Popup'
+import InAppBrowser from '../../InAppBrowser'
+import { isFlagshipApp } from 'cozy-device-helper'
 findKonnectorPolicy.mockImplementation(() => ({
   fetchContractSynchronizationUrl,
   refreshContracts
 }))
 
+const mockKonnector = { slug: 'mockkonnector' }
+const mockAccount = {}
+
+const setup = () => {
+  const client = new CozyClient({})
+  return render(
+    <AppLike client={client}>
+      <BIContractActivationWindow
+        konnector={mockKonnector}
+        account={mockAccount}
+      />
+    </AppLike>
+  )
+}
+
 describe('BIContractActivationWindow', () => {
-  const mockKonnector = { slug: 'mockkonnector' }
-  const mockAccount = {}
-  const setup = () => {
-    const client = new CozyClient({})
-    const root = render(
-      <AppLike client={client}>
-        <BIContractActivationWindow
-          konnector={mockKonnector}
-          account={mockAccount}
-        />
-      </AppLike>
-    )
-    return { root }
-  }
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  afterAll(() => {
+    jest.resetAllMocks()
+  })
 
   it('should display popup with url from policy and update contract after popup is closed', async () => {
+    isFlagshipApp.mockImplementation(() => false)
     fetchContractSynchronizationUrl.mockResolvedValue('bi url')
-    let root
+    const { getByRole } = setup()
     await act(async () => {
-      const res = setup()
-      root = res.root
-      const button = root.getByRole('button')
+      const button = getByRole('button')
       await waitFor(() => {
         expect(button.getAttribute('class')).not.toContain('Mui-disabled')
```

Comment:
nit: `expect(getByRole('button').getAttribute('class')).not.toContain('Mui-disabled')`